src/test/java/org/springframework/integration/aws/outbound/KinesisProducingMessageHandlerTests.java

```diff
@@ -0,0 +1,237 @@
+		try {
+			this.kinesisSendChannel.send(message);
+		}
+		catch (Exception e) {
+			assertThat(e).isInstanceOf(MessageHandlingException.class);
+			assertThat(e.getCause()).isInstanceOf(IllegalStateException.class);
+			assertThat(e.getMessage()).contains("'stream' must not be null for sending a Kinesis record");
+		}
+
+		this.kinesisMessageHandler.setStream("foo");
+		try {
+			this.kinesisSendChannel.send(message);
+		}
+		catch (Exception e) {
+			assertThat(e).isInstanceOf(MessageHandlingException.class);
+			assertThat(e.getCause()).isInstanceOf(IllegalStateException.class);
+			assertThat(e.getMessage()).contains("'partitionKey' must not be null for sending a Kinesis record");
+		}
+
+		message = MessageBuilder.fromMessage(message)
+				.setHeader(AwsHeaders.PARTITION_KEY, "fooKey")
+				.setHeader(AwsHeaders.SEQUENCE_NUMBER, "10")
+				.build();
+
+		this.kinesisSendChannel.send(message);
+
+		ArgumentCaptor<PutRecordRequest> putRecordRequestArgumentCaptor =
+				ArgumentCaptor.forClass(PutRecordRequest.class);
+		verify(this.amazonKinesis).putRecordAsync(putRecordRequestArgumentCaptor.capture(),
+				any(AsyncHandler.class));
+
+		PutRecordRequest putRecordRequest = putRecordRequestArgumentCaptor.getValue();
+
+		assertThat(putRecordRequest.getStreamName()).isEqualTo("foo");
+		assertThat(putRecordRequest.getPartitionKey()).isEqualTo("fooKey");
+		assertThat(putRecordRequest.getSequenceNumberForOrdering()).isEqualTo("10");
+		assertThat(putRecordRequest.getExplicitHashKey()).isNull();
+		assertThat(putRecordRequest.getData()).isEqualTo(ByteBuffer.wrap("message".getBytes()));
+
+		message = MessageBuilder.fromMessage(message)
+				.setHeader(AwsHeaders.PARTITION_KEY, "fooKey")
+				.setHeader(AwsHeaders.SEQUENCE_NUMBER, "10")
+				.build();
+
+		this.kinesisSendChannel.send(message);
+
+		Message<?> failed = this.errorChannel.receive(10000);
+		AwsSendFailureException putRecordFailure = (AwsSendFailureException) failed.getPayload();
+		assertThat(putRecordFailure.getCause().getMessage()).isEqualTo("putRecordRequestEx");
+		assertThat(((PutRecordRequest) putRecordFailure.getRecord()).getStreamName()).isEqualTo("foo");
+		assertThat(((PutRecordRequest) putRecordFailure.getRecord()).getPartitionKey()).isEqualTo("fooKey");
+		assertThat(((PutRecordRequest) putRecordFailure.getRecord()).getSequenceNumberForOrdering()).isEqualTo("10");
+		assertThat(((PutRecordRequest) putRecordFailure.getRecord()).getExplicitHashKey()).isNull();
+		assertThat(((PutRecordRequest) putRecordFailure.getRecord())
+				.getData()).isEqualTo(ByteBuffer.wrap("message".getBytes()));
+
+		message = new GenericMessage<>(new PutRecordsRequest()
+				.withStreamName("myStream")
+				.withRecords(new PutRecordsRequestEntry()
+						.withData(ByteBuffer.wrap("test".getBytes()))
+						.withPartitionKey("testKey")));
+
+		this.kinesisSendChannel.send(message);
+
+		ArgumentCaptor<PutRecordsRequest> putRecordsRequestArgumentCaptor =
```

Comment:
I think would be better to use `KinesisMessageHandler.setOutputChannel()` to track success for the send via our `onSuccess()` callback.